src/Console/DeployCommand.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Laravel\Nightwatch\Console;
+
+use Exception;
+use Illuminate\Console\Command;
+use Illuminate\Http\Client\Factory as HttpFactory;
+use SensitiveParameter;
+use Symfony\Component\Console\Attribute\AsCommand;
+
+use function config;
+use function now;
+
+/**
+ * @internal
+ */
+#[AsCommand(name: 'nightwatch:deploy', description: 'Notice the Nightwatch agent that a new deployment has been made.')]
+final class DeployCommand extends Command
+{
+    /**
+     * @var string
+     */
+    protected $signature = 'nightwatch:deploy';
+
+    protected $hidden = true;
+
+    /**
+     * @var string
+     */
+    protected $description = 'Notice the Nightwatch agent that a new deployment has been made.';
+
+    public function __construct(
+        private HttpFactory $http,
+        private ?string $baseUrl,
+        #[SensitiveParameter] private ?string $token,
+    ) {
+        parent::__construct();
+    }
+
+    public function handle(): void
+    {
+        $tag = config('nightwatch.deployment') ?? '';
+
+        if (! $this->baseUrl) {
+            $this->error('No Nightwatch base URL configured.');
+
+            return;
+        }
```

Comment:
In the agent we define the base URL like this, rather than exposing it via config, as it's not something users should ever need to configure.

```suggestion
        $baseUrl ??= $_SERVER['NIGHTWATCH_BASE_URL'] ?? 'https://nightwatch.laravel.com';
```